jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/XGBoost.java

```diff
@@ -217,6 +212,45 @@ public static Booster train(
     return booster;
   }
 
+  static boolean judgeIfTrainingOnTrack(
+          Map<String, Object> params, int earlyStoppingRounds, float[][] metrics, int iter) {
+    boolean metricsExpectedDirection = getMetricsExpectedDirection(params, earlyStoppingRounds);
+    boolean onTrack = false;
+    float[] criterion = metrics[metrics.length - 1];
+    for (int shift = 0; shift < Math.min(iter, earlyStoppingRounds) - 1; shift++) {
+      onTrack |= metricsExpectedDirection ?
+              criterion[iter - shift] >= criterion[iter - shift - 1] :
+              criterion[iter - shift] <= criterion[iter - shift - 1];
+    }
+    return onTrack;
+  }
+
+  private static String getReversedDirection(Map<String, Object> params) {
+    String reversedDirection = null;
+    if (params.get("metrics_expected_direction").equals("asc")) {
+      reversedDirection = "descending";
+    } else if (params.get("metrics_expected_direction").equals("desc")) {
+      reversedDirection = "ascending";
+    }
+    return reversedDirection;
+  }
+
+  private static boolean getMetricsExpectedDirection(
+      Map<String, Object> params, int earlyStoppingRound) {
+    boolean metricsExpectedDirection = false;
+    if (earlyStoppingRound > 0) {
+      if (params.get("metrics_expected_direction").equals("asc")) {
+        metricsExpectedDirection = true;
+      } else if (params.get("metrics_expected_direction").equals("desc")) {
+        metricsExpectedDirection = false;
```

Comment:
The code here is a little confused. If we use boolean to represent ```asc``` or ```desc```, it's better to rename ```metricsExpectedDirection``` as ```metricsExpectedAsc```.